plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -430,6 +439,16 @@ public IcebergMetadata(
         this.metastoreFactory = requireNonNull(metastoreFactory, "metastoreFactory is null");
         this.addFilesProcedureEnabled = addFilesProcedureEnabled;
         this.allowedExtraProperties = requireNonNull(allowedExtraProperties, "allowedExtraProperties is null");
+
+        if (metadataParallelism == 1) {
+            this.executorService = directExecutor();
+        }
+        else {
+            this.executorService = Executors.newFixedThreadPool(metadataParallelism, new ThreadFactoryBuilder()
+                    .setNameFormat("iceberg-metadata-pool-%d")
+                    .setDaemon(true) // Optional: Makes threads daemon threads.
+                    .build());
```

Comment:
Why is the logic different from another metadata executor for listing tables?

Reply:
I was follow this where in recent or they changed it to a similar thing https://github.com/trinodb/trino/blob/4718011a39f700ab3c8dbe66747c37a9398922f8/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadataFactory.java#L111

Can you give me an example for another metadata executer so I will follow the same logic

And the use of ThreadFactoryBuilder is required because of the maven checks tests that require that each thread will have a name format